pkg/netutils/netutils.go

```diff
@@ -277,3 +277,23 @@ func isExist(mmap map[string]bool, key string) bool {
 	}
 	return false
 }
+
+// CalculateTimeout calculate the timeout according to the fileLength and the min rate of network.
+//
+// The 0 will be returned when both minRate and defaultMinRate both are <=0.
+func CalculateTimeout(fileLength int64, minRate int, defaultMinRate int) time.Duration {
```

Comment:
And please add a unit test for this function.

Reply:
Done.